apps/studio/components/interfaces/Integrations/Vault/Secrets/SecretsManagement.tsx

```diff
@@ -81,26 +88,21 @@ export const SecretsManagement = () => {
                   : []
               }
             />
-            <div className="md:w-44">
-              <Listbox size="tiny" value={selectedSort} onChange={setSelectedSort}>
-                <Listbox.Option
-                  id="updated_at"
-                  className="max-w-[180px]"
-                  value="updated_at"
-                  label="Sort by updated at"
-                >
+            <Select_Shadcn_ value={selectedSort} onValueChange={setSelectedSort}>
+              <SelectTrigger_Shadcn_ size="tiny" className="md:w-44">
+                <SelectValue_Shadcn_ asChild>
+                  <>Sort by {selectedSort}</>
+                </SelectValue_Shadcn_>
+              </SelectTrigger_Shadcn_>
+              <SelectContent_Shadcn_>
+                <SelectItem_Shadcn_ value="updated_at" className="text-xs">
                   Updated at
-                </Listbox.Option>
-                <Listbox.Option
-                  id="name"
-                  className="max-w-[180px]"
-                  value="name"
-                  label="Sort by name"
-                >
+                </SelectItem_Shadcn_>
+                <SelectItem_Shadcn_ value="name" className="text-xs">
                   Name
-                </Listbox.Option>
-              </Listbox>
-            </div>
+                </SelectItem_Shadcn_>
+              </SelectContent_Shadcn_>
+            </Select_Shadcn_>
```

Comment:
I'm not sure it matters but I do want to point out that the select dropdown icon is different between these two implementations

Reply:
additionally, I noticed that when TanStack dev tools are open, opening this select causes a layout shift, so if you ever see that and wonder why, it's not something you need to worry about debugging

Reply:
the selected check icon does look big. I think this needs `portal=false` to fix the layout issue?

![screenshot-2025-07-28-at-17 13 11](https://github.com/user-attachments/assets/da6cfb14-3544-4130-94b3-19328c83a0d6)

Reply:
hmm, that appears to be off. this is what I am seeing on me end:

<img width="867" height="562" alt="image" src="https://github.com/user-attachments/assets/75616458-8ecf-4dc9-b443-91db1371fbc3" />

that layout issue would appear to be consistent with the old vault sort select